sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -638,21 +638,25 @@ case class GpuCast(
 
     // determine which values are valid bool strings
     withResource(ColumnVector.fromStrings(boolStrings: _*)) { boolStrings =>
-      withResource(input.contains(boolStrings)) { validBools =>
-        // in ansi mode, fail if any values are not valid bool strings
-        if (ansiEnabled) {
-          withResource(validBools.all()) { isAllBool =>
-            if (!isAllBool.getBoolean) {
-              throw new IllegalStateException(GpuCast.INVALID_INPUT_MESSAGE)
+      withResource(input.strip()) { stripped =>
+        withResource(stripped.lower()) { lower =>
```

Comment:
technically, JDK roundtrips the string toUpper->toLower  in order to normalize the string for case-insensitive comparison https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/String.java#L1941 but we should be ok with the target values "true" "false".

Reply:
Spark has [custom code to do the conversion](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala#L75-L77).  Spark isn't using the JDK to do the cast.

Reply:
Thanks for pointing this out. Let us say Spark does not use equalsIgnoreCase :)